tools/ReleaseProcess/UploadArtifactsToGitHub.ps1

```diff
@@ -0,0 +1,444 @@
+        #>
+        [Parameter(Mandatory)]
+        [string]
+        $WorkDir
+    )
+
+    # Assume Az CLI is installed & logged in.
+    $pat = $(az keyvault secret show -n IotEdge1-PAT-msazure --vault-name edgebuildkv | ConvertFrom-Json)
+    $pat = $pat.value;
+
+    $encoded64Pat = [Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes(':'+$pat))
+    $header = @{
+        "Authorization" = "Basic $encoded64Pat"
+    }
+
+    # Get all the artifact names
+    $response = $(Invoke-WebRequest -Uri "https://dev.azure.com/msazure/One/_apis/build/builds/$BuildId/artifacts?api-version=6.0" -Headers $header)
+    $content = ($response.Content | ConvertFrom-Json).value
+
+    $outputDir = "$(Join-Path -Path $WorkDir -ChildPath 'output')\"
+    New-Item -ItemType Directory -Force -Path $outputDir
+    $WorkDir = "$(Join-Path -Path $WorkDir -ChildPath 'IE')\"
+    New-Item -ItemType Directory -Force -Path $WorkDir
+
+    $artifactFinalists = @();
+
+    foreach ($artifact in $content)
+    {
+        $artifactName = $artifact.name
+        $artifactUrl = $artifact.resource.downloadUrl
+        $artifactPath = "$WorkDir$artifactName"
+        $artifactExtension = ".zip"
+
+        # Download and Expand each artifact
+        Retry-Command -ScriptBlock {
+            echo "Downloading $artifactName"
+            Invoke-WebRequest -Uri $artifactUrl -Headers $header -OutFile "$artifactPath$artifactExtension" | Out-Null
+            Expand-Archive -Path "$artifactPath$artifactExtension" -DestinationPath $WorkDir -Force
+        }
+
+        # Each artifact is a directory, fetch the packages within it.
+        $packages = $(Get-ChildItem -Path $artifactPath/* -Recurse `
+            -Include "*.deb", "*.rpm" `
+            -Exclude "*.src*", "*dev*", "*dbg*", "*debug*" `
+            | where { ! $_.PSIsContainer })
+
+        # Within each directory, rename the artifacts
+        $component,$os,$suffix = $artifactName.split('-')
+
+        # CentOs7 packages do not need renaming.
+        if ($os -eq "centos7")
+        {
+            echo "Skip renaming :"
+            echo $($packages.FullName)
+            $artifactFinalists += $packages;
+            continue;
+        }
+
+        # Ranaming the artifacts
+        foreach ($package in $packages)
+        {
+            echo "Processing : $($package.FullName)"
+            $name,$version,$arch,$suffix = $package.Name.split("_")
+            $arch,$ext,$suffix = $arch.split(".")
+
+            # Reconstruct the new name from the segments. 
+            $finalName = @($name, $version, $os, $arch) -join '_'
+            $finalName = "$finalName.$ext"
+            $newPath = $(Join-Path -Path "$($package.Directory)" -ChildPath "$finalName")
+
+            # Rename
+            Rename-Item -Path $package.FullName -NewName $newPath
+
+            # Record renamed files
+            $artifactFinalists += $(Get-Item $newPath)
+        }
+
+        echo ""
```

Comment:
What is the difference between echo and WriteOutput?

Reply:
In powershell, `echo` is alised to `WriteOutput`.

Reply:
I read somewhere it's a good practice to avoid aliases inside scripts (possible security problem if alias points to something completely different in a user's environment). But that's just a rumor...